packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -266,20 +266,28 @@ export default createRule<Options, MessageId>({
       if (isStrictCompilerOptionEnabled(compilerOptions, 'strictNullChecks')) {
         const UNDEFINED = ts.TypeFlags.Undefined;
         const NULL = ts.TypeFlags.Null;
+        const isComparable = (type: ts.Type, flag: ts.TypeFlags): boolean => {
+          // Allow comparison to `any`, `unknown` or a naked type parameter.
+          flag |=
+            ts.TypeFlags.Any |
+            ts.TypeFlags.Unknown |
+            ts.TypeFlags.TypeParameter;
+
+          // Allow loose comparison to nullish values.
+          if (node.operator === '==' || node.operator === '!=') {
+            flag |= NULL | UNDEFINED;
+          }
 
-        const NULLISH =
-          node.operator === '==' || node.operator === '!='
-            ? NULL | UNDEFINED
-            : NULL;
+          return isTypeFlagSet(type, flag);
```

Comment:
I decided to drop the usage of the `isReceiver` option to expose `Any` and `Unknown` types usage.